src/EnergyEfficientAgent.cpp

```diff
@@ -215,6 +215,19 @@ namespace geopm
         m_freq_governor->adjust_platform(m_target_freq);
     }
 
+    bool EnergyEfficientAgent::is_region_boundary(const struct m_region_info_s &last_region_info, const struct m_region_info_s &current_region_info)
+    {
+        bool ret = false;
+        if (last_region_info.hash != current_region_info.hash || // region hash change
+            (last_region_info.hash == current_region_info.hash &&
+             last_region_info.count != current_region_info.count) || // same hash, new count
+            (last_region_info.hash == current_region_info.hash &&
+             last_region_info.hint != current_region_info.hint)) { // same hash, new hint
```

Comment:
With this change in place, does this mean that if MPI is nested within a user-marked region that the nested MPI is treated as a wholly separate region?

It seems that way for the last case in the if where the hash is the same but the hint has changed.